crates/node/types/src/lib.rs

```diff
@@ -41,6 +41,15 @@ pub trait NodeTypes: Send + Sync + Unpin + 'static {
     type ChainSpec: EthChainSpec;
 }
 
+/// Dynamic version of [`NodeTypes`] with node primitives block being [`reth_primitives::Block`].
+pub trait NodeTypesDyn: Send + Sync + Unpin + 'static {
+    /// The node's primitive types, defining basic operations and structures.
+    fn primitives(&self) -> &(dyn NodePrimitives<Block = reth_primitives::Block> + 'static);
+
+    /// The type used for configuration of the EVM.
+    fn chain_spec(&self) -> Box<dyn EthChainSpec + 'static>;
+}
+
```

Comment:
do we need this now?

Reply:
No, it’s just experimental for Node types, can remove for now. Tried to understand better way for components abstraction